xata/app/routes/index.tsx

```diff
@@ -0,0 +1,134 @@
+import type { FC } from "react";
+import type { ActionArgs } from "@remix-run/node";
+import { json } from "@remix-run/node";
+import { Form, useFetcher, useLoaderData } from "@remix-run/react";
+
+import type { RemixWithXataExampleRecord } from "~/lib/xata.codegen.server";
+import { getXataClient } from "~/lib/xata.codegen.server";
+
+export const LINKS = [
+  {
+    description: "Everything you need to know about Xata APIs and tools.",
+    title: "Xata Docs",
+    url: "https://docs.xata.io",
+  },
+  {
+    description: "In case you need to check some Remix specifics.",
+    title: "Remix Docs",
+    url: "https://remix.run/docs",
+  },
+  {
+    description:
+      "Maintain your flow by managing your Xata Workspace without ever leaving VS Code.",
+    title: "Xata VS Code Extension",
+    url: "https://marketplace.visualstudio.com/items?itemName=xata.xata",
+  },
+  {
+    description: "Get help. Offer help. Show us what you built!",
+    title: "Xata Discord",
+    url: "https://xata.io/discord",
+  },
+];
+
+type TaskComponent = FC<
+  Pick<RemixWithXataExampleRecord, "id" | "title" | "url" | "description">
+>;
+export const loader = async () => {
+  const xata = getXataClient();
+  const links = await xata.db.remix_with_xata_example.getAll();
+
+  return json(links);
+};
+
+export const action = async ({ request }: ActionArgs) => {
+  const xata = getXataClient();
+  const { intent, id } = Object.fromEntries(await request.formData());
+
+  if (intent === "delete" && typeof id === "string") {
+    await xata.db.remix_with_xata_example.delete(id);
+
+    return json({
+      message: "delete: success",
+      data: null,
+    });
+  }
+
+  if (intent === "create") {
+    const newItem = await xata.db.remix_with_xata_example.create(LINKS);
+
+    return json({
+      message: "create: success",
+      data: newItem,
+    });
+  }
+
+  return json({
+    message: "no action performed",
+    data: null,
+  });
+};
+
+const Task: TaskComponent = ({ id, title, url, description }) => {
+  const fetcher = useFetcher();
+
+  return fetcher.submission ? null : (
+    <li key={url}>
+      <a href={url ?? ""} rel="noopener noreferrer" target="_blank">
+        {title}
+      </a>
+      <p>{description}</p>
+      <fetcher.Form method="put">
+        <input type="hidden" name="intent" value="delete" />
+        <button type="submit" name="id" value={id}>
```

Comment:
It's a best practice to put `intent` on the `button` and any other data you want to pass in a hidden `input`

```suggestion
        <input type="hidden" name="id" value={id} />
        <button type="submit" name="intent" value="delete">
```